### ISSUE TITLE
dbx deploy fails due to mlflow experiment not found

### ISSUE THREAD

**Issue by zermelozf** [databrickslabs/dbx#385]
## Expected Behavior
`dbx deploy --environment=default` succeeds

## Current Behavior
The command returns 
`mlflow.exceptions.RestException: INVALID_PARAMETER_VALUE: Experiment with id '2170254243754186' does not exist.`

## Steps to Reproduce (for bugs)
Follow the instructions at https://docs.gcp.databricks.com/dev-tools/ide-how-to.html#run-with-dbx

## Context
Trying to set up dbx for the first time.

## Your Environment
mac os m1 2021 with macos Monterey 12.5

* dbx version used: DataBricks eXtensions aka dbx, version ~> 0.6.11
* Databricks Runtime version: Version 0.17.1

**Comment by renardeinside** [comment]
hi @zermelozf , 
could you please provide full stack trace?

**Comment by zermelozf** [comment]
Sure, here it is:

```
dbx deploy --environment=default
[dbx][2022-08-13 22:46:37.005] Starting new deployment for environment default
[dbx][2022-08-13 22:46:37.006] Using profile provided from the project file
[dbx][2022-08-13 22:46:37.006] Found auth config from provider ProfileEnvConfigProvider, verifying it
[dbx][2022-08-13 22:46:37.007] Found auth config from provider ProfileEnvConfigProvider, verification successful
[dbx][2022-08-13 22:46:37.007] Profile DEFAULT will be used for deployment
Traceback (most recent call last):
  File "/opt/homebrew/anaconda3/envs/databricks/bin/dbx", line 8, in <module>
    sys.exit(cli())
  File "/opt/homebrew/anaconda3/envs/databricks/lib/python3.9/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/opt/homebrew/anaconda3/envs/databricks/lib/python3.9/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/opt/homebrew/anaconda3/envs/databricks/lib/python3.9/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/homebrew/anaconda3/envs/databricks/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/homebrew/anaconda3/envs/databricks/lib/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/opt/homebrew/anaconda3/envs/databricks/lib/python3.9/site-packages/dbx/commands/deploy.py", line 143, in deploy
    api_client = prepare_environment(environment)
  File "/opt/homebrew/anaconda3/envs/databricks/lib/python3.9/site-packages/dbx/utils/common.py", line 38, in prepare_environment
    MlflowStorageConfigurationManager.prepare(info)
  File "/opt/homebrew/anaconda3/envs/databricks/lib/python3.9/site-packages/dbx/api/storage/mlflow_based.py", line 42, in prepare
    cls._setup_experiment(properties)
  File "/opt/homebrew/anaconda3/envs/databricks/lib/python3.9/site-packages/dbx/api/storage/mlflow_based.py", line 53, in _setup_experiment
    experiment: Optional[Experiment] = mlflow.get_experiment_by_name(properties.workspace_dir)
  File "/opt/homebrew/anaconda3/envs/databricks/lib/python3.9/site-packages/mlflow/tracking/fluent.py", line 1042, in get_experiment_by_name
    return MlflowClient().get_experiment_by_name(name)
  File "/opt/homebrew/anaconda3/envs/databricks/lib/python3.9/site-packages/mlflow/tracking/client.py", line 566, in get_experiment_by_name
    return self._tracking_client.get_experiment_by_name(name)
  File "/opt/homebrew/anaconda3/envs/databricks/lib/python3.9/site-packages/mlflow/tracking/_tracking_service/client.py", line 226, in get_experiment_by_name
    return self.store.get_experiment_by_name(name)
  File "/opt/homebrew/anaconda3/envs/databricks/lib/python3.9/site-packages/mlflow/store/tracking/rest_store.py", line 365, in get_experiment_by_name
    raise e
  File "/opt/homebrew/anaconda3/envs/databricks/lib/python3.9/site-packages/mlflow/store/tracking/rest_store.py", line 351, in get_experiment_by_name
    response_proto = self._call_endpoint(GetExperimentByName, req_body)
  File "/opt/homebrew/anaconda3/envs/databricks/lib/python3.9/site-packages/mlflow/store/tracking/rest_store.py", line 57, in _call_endpoint
    return call_endpoint(self.get_host_creds(), endpoint, method, json_body, response_proto)
  File "/opt/homebrew/anaconda3/envs/databricks/lib/python3.9/site-packages/mlflow/utils/rest_utils.py", line 274, in call_endpoint
    response = verify_rest_response(response, endpoint)
  File "/opt/homebrew/anaconda3/envs/databricks/lib/python3.9/site-packages/mlflow/utils/rest_utils.py", line 200, in verify_rest_response
    raise RestException(json.loads(response.text))
mlflow.exceptions.RestException: INVALID_PARAMETER_VALUE: Experiment with id '2170254243754186' does not exist.
```

**Comment by renardeinside** [comment]
hi @zermelozf , 
it seems to me that you're using an old version of `dbx`. Please upgrade to the latest 0.7.0 (or at least to 0.6.12).

**Comment by frida-ah** [comment]
hi @renardeinside I had the same issue mentioned here. I upgraded to dbx 0.7.0 and now the error looks like this:
RestException: INVALID_PARAMETER_VALUE: Experiment with id '2624352622693299' does not exist.
It only happens if you deploy a job for the first time. Deploying changes to an existing job works fine.

**Comment by renardeinside** [comment]
hi @frida-ah , 
what's the MLflow version you're using? I'm asking because I'm not running into this issue in any of the tests

**Comment by renardeinside** [comment]
could you please also verify that you have correct [databricks profile configured as in Step 3 point 4 of the public doc](https://docs.gcp.databricks.com/dev-tools/ide-how-to.html#step-3-install-the-code-samples-dependencies)?

if it's still the case, please provide the deploy command with `dbx deploy --debug` option (please feel free to omit the host url)? 
Really curious where is this coming from.

**Comment by frida-ah** [comment]
Hi @renardeinside I don't have mlflow in my requirements.txt. I can also confirm that I have the correct databricks profile configured in the deployment.json file as such:

{
  "environments": {
    "default": {
      "profile": "DEFAULT",
      "workspace_dir": "/Shared/dbx/projects/<project_name>/<...>",
      "artifact_location": "dbfs:/Shared/dbx/projects/<project_name>/<...>"
    }
  }
}

dbx deploy --environment default --deployment-file=conf/deployment.json --jobs=<job_name>

I have fixed the issue using a workaround - sorry I didn't have more time to invest in this. I created an artifact manually through the UI in the location where the artifact should be. Then I deleted it. And then the artifact was created again through the IDE and GitHub Actions. 

I think the issue is with Databricks having a bug when creating an artifact for the first time.
